src/components/DefaultRedirectionSection.jsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react'
+
+import { useI18n } from 'cozy-ui/transpiled/react/I18n'
+import {
+  useQuery,
+  hasQueryBeenLoaded,
+  deconstructRedirectLink
+} from 'cozy-client'
+
+import Select from 'components/Select'
+import { buildAppsQuery } from 'lib/queries'
+import { useWebviewIntent } from 'cozy-intent'
+
+const EXCLUDED_SLUGS = ['settings', 'store']
+
+const formatOptions = (appsResult, t) => {
+  if (!hasQueryBeenLoaded(appsResult)) return []
+
+  return appsResult.data
+    .filter(({ slug }) => !EXCLUDED_SLUGS.includes(slug))
+    .map(({ slug, name }, index) => {
+      if (slug === 'home') {
+        return {
+          index: -1, // to set 'home' before everything else when sorting
+          slug,
+          name: t('ProfileView.default_redirection.app_list')
+        }
+      } else {
+        return {
+          index,
+          slug,
+          name
+        }
+      }
+    })
+    .sort((a, b) => a.index - b.index)
+}
+
+const getSelectedOption = (defautRedirection, options, t) => {
+  try {
+    const { slug } = deconstructRedirectLink(defautRedirection)
+    const { name } = options.find(option => option.slug === slug)
+
+    return {
+      value: slug,
+      label: name
+    }
+  } catch {
+    return {
+      value: 'home',
+      label: t('ProfileView.default_redirection.app_list')
+    }
+  }
+}
+
+const formatDefaultRedirection = slug => `${slug}/`
+
+const DefaultRedirectionSection = props => {
+  const { t } = useI18n()
+  const webviewIntent = useWebviewIntent()
+  const appsQuery = buildAppsQuery()
+  const appsResult = useQuery(appsQuery.definition, appsQuery.options)
+
+  const options = formatOptions(appsResult, t)
+
+  const { fields, onChange } = props
+  const fieldProps = {
+    ...fields.default_redirection,
+    title: t('ProfileView.default_redirection.title'),
+    label: t(`ProfileView.default_redirection.label`)
+  }
+  const selectedSlug = getSelectedOption(
+    fields.default_redirection.value,
+    options,
+    t
+  )
+  const fieldName = 'default_redirection'
+  return (
+    <div>
+      <Select
+        name={fieldName}
+        options={options.map(app => {
+          return {
+            value: app.slug,
+            label: app.name
+          }
+        })}
+        fieldProps={fieldProps}
+        value={selectedSlug}
+        onChange={sel => {
+          const newDefaultRedirection = formatDefaultRedirection(sel.value)
+          onChange(fieldName, newDefaultRedirection)
+          webviewIntent.call('setDefaultRedirection', newDefaultRedirection)
```

Comment:
Should we condition this call to `isFlagshipApp()`?

Reply:
Yes I think